### PR TITLE
Textfield mode variants

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Changed mode variant class and added variables for dark mode

**Solving issue**  
Fixes: [AB#3114](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/3114)

### CODE REVIEW

File: tegel/src/components/textfield/textfield-theme-vars.scss [hunk 1]
```diff
@@ -99,16 +106,21 @@
...
   --sdds-textfield-border-bottom: var(--sdds-grey-600);
   --sdds-textfield-border-bottom-hover: var(--sdds-grey-400);
   --sdds-textfield-border-bottom-success: var(--sdds-grey-400);
   --sdds-textfield-border-bottom-error: var(--sdds-negative);
 
-    // Read only
-    --sdds-textfield-icon-read-only-color: var(--sdds-grey-100);
-    --sdds-textfield-icon-read-only-label-color: var(--sdds-grey-50);
+  // Read only
+  --sdds-textfield-icon-read-only-color: var(--sdds-grey-100);
```
Comment (nathalielindqvist) [inline tegel/src/components/textfield/textfield-theme-vars.scss:116]: Read only icon color is Grey600 in Figma from what I can see.
image

File: tegel/src/components/textfield/textfield-theme-vars.scss [hunk 2]
```diff
@@ -99,16 +106,21 @@
...
+  --sdds-textfield-icon-read-only-label-color: var(--sdds-grey-50);
```
Comment (nathalielindqvist) [inline tegel/src/components/textfield/textfield-theme-vars.scss:117]: Cannot see read-only label in Figma or in Deploy Preview, is it supposed to be visible?

File: tegel/src/components/textfield/textfield.tsx [hunk 3]
```diff
@@ -103,7 +103,7 @@ export class Textfield {
         }
         ${this.disabled ? 'sdds-form-textfield-disabled' : ''}
         ${this.readonly ? 'sdds-form-textfield-readonly' : ''}
-        ${this.modeVariant === 'primary' ? 'sdds-on-white-bg' : ''}
+        sdds-mode-variant-${this.modeVariant}
         ${this.size === 'md' ? 'sdds-form-textfield-md' : ''}
         ${this.size === 'sm' ? 'sdds-form-textfield-sm' : ''}
         ${
```
Comment (mJarsater) [inline tegel/src/components/textfield/textfield.tsx:109]: Really not part of this PR, but could we change the svg out for own `<sdds-icon name="edit_inactive">`?

**Comment by mJarsater** [comment]
Really not part of this PR, but could we change the svg out for own `<sdds-icon name="edit_inactive">`?

**Review by mJarsater — APPROVED** [review]

**Comment by EmelieLitwin** [comment]
@charlottelorensson Can you do a review now that borders and readonly is hopefully fixed?

**Comment by charlottelorensson** [comment]
Not sure if it belongs in this review but the disabled state does not seem to work properly when switching between primary and secondary state (both for light and dark mode)? It also shows some lightly faded divider that is not in Figma.

**Comment by EmelieLitwin** [comment]
> Not sure if it belongs in this review but the disabled state does not seem to work properly when switching between primary and secondary state (both for light and dark mode)? It also shows some lightly faded divider that is not in Figma.

Good find Charlotte! I updated the disabled backgrounds but I can't find the divider issues. Could you maybe take a screenshot 🙏

**Comment by charlottelorensson** [comment]
> > Not sure if it belongs in this review but the disabled state does not seem to work properly when switching between primary and secondary state (both for light and dark mode)? It also shows some lightly faded divider that is not in Figma.
> 
> Good find Charlotte! I updated the disabled backgrounds but I can't find the divider issues. Could you maybe take a screenshot 🙏

I can't see the divider issue there anymore :) It was when hovering on the disabled but maybe the backgrounds fixed it then? :)
